### PR TITLE
[DOC] Update Configure the client and span profiles doc

### PR DESCRIPTION
**Pull request by knylander-grafana**
This PR does two things: 
* Corrects capitalization for Grafana Agent and makes some minor edits outlined in #2771. 
* Updates the Span Profiles doc and adds a link to the Traces to profiles doc in the Pyroscope documentation

Fixes https://github.com/grafana/pyroscope/issues/2771

### CODE REVIEW

File: docs/sources/configure-client/go-span-profiles.md [hunk 1]
```diff
@@ -7,38 +7,52 @@ weight: 100
...
+
+This shift enables a more granular view of performance, enhancing the utility of profiles by linking them directly with traces for a comprehensive understanding of application behavior. As a result, engineering teams can more efficiently identify and address performance bottlenecks.
+
+To learn more about Span Profiles, refer to [Combining tracing and profiling for enhanced observability: Introducing Span Profiles](/blog/2024/02/06/combining-tracing-and-profiling-for-enhanced-observability-introducing-span-profiles/).
+
 ![span-profiles screenshot](https://grafana.com/static/img/docs/tempo/profiles/tempo-profiles-Span-link-profile-data-source.png)
 
+To use Span Profiles, you need to:
+
+* [Configure Pyroscope to send profiling data](./configure-client)
```
Comment (knylander-grafana) [inline docs/sources/configure-client/go-span-profiles.md:22]:
```suggestion
* [Configure Pyroscope to send profiling data]({{< relref "./" >}})
```

File: docs/sources/configure-client/go-span-profiles.md [hunk 2]
```diff
@@ -64,37 +78,25 @@ func main() {
...
-1. Select a Pyroscope data source from the Data source drop-down.
-
-2. Optional: Choose any tags to use in the query. If left blank, the default values of service.name and service.namespace are used.
-
-The tags you configure must be present in the spans attributes or resources for a trace to profiles span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source doesn’t allow using dots in labels. In that case you can for example remap service.name to service_name.
+## View the span profiles in Grafana or Grafana Cloud
 
-3. Select one or more profile types to use in the query. Select the drop-down and choose options from the menu.
+To view the span profiles in Grafana Tempo, you need to have a Grafana instance with a Tempo data source configured to link trace spans and profiles.
+Refer to the configuration documentation for [Grafana](/docs/grafana/<GRAFANA_VERSION>/datasources/tempo/configure-tempo-data-source) or [Grafana Cloud]([/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/configure-tempo-data-source).
```
Comment (knylander-grafana) [inline docs/sources/configure-client/go-span-profiles.md:92]:
```suggestion
Refer to the configuration documentation for [Grafana](/docs/grafana/<GRAFANA_VERSION>/datasources/tempo/configure-tempo-data-source) or [Grafana Cloud](/docs/grafana-cloud/connect-externally-hosted/data-sources/tempo/configure-tempo-data-source).
```

File: docs/sources/configure-client/go-span-profiles.md [hunk 3]
```diff
@@ -64,37 +78,25 @@ func main() {
...
-Refer to the [data source configuration documentation](/docs/grafana/datasources/tempo/configure-tempo-data-source) to see how to configure the visualization to link trace spans with profiles.
-
-To use a simple configuration, follow these steps:
-
-1. Select a Pyroscope data source from the Data source drop-down.
-
-2. Optional: Choose any tags to use in the query. If left blank, the default values of service.name and service.namespace are used.
-
-The tags you configure must be present in the spans attributes or resources for a trace to profiles span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source doesn’t allow using dots in labels. In that case you can for example remap service.name to service_name.
+## View the span profiles in Grafana or Grafana Cloud
```
Comment (knylander-grafana) [inline docs/sources/configure-client/go-span-profiles.md:90]: We have the full content for Traces to profiles in the Pyroscope documentation. I removed the content from the Traces to profiles and linked to the page instead. 

Doc is here: https://grafana.com/docs/pyroscope/next/view-and-analyze-profile-data/profile-tracing/traces-to-profiles/ 

We can make it live. It's currently in /next. Not sure why it isn't live.

File: docs/sources/configure-client/go-span-profiles.md [hunk 4]
```diff
@@ -7,38 +7,52 @@ weight: 100
 
 # Span profiles with Traces to profiles for Go
```
Comment (knylander-grafana) [inline docs/sources/configure-client/go-span-profiles.md:9]: The blog post about span profiles has some great content, so I used a few of the paragraphs for the introduction and linked to the blog post.

Blog post: https://grafana.com/blog/2024/02/06/combining-tracing-and-profiling-for-enhanced-observability-introducing-span-profiles/

File: docs/sources/configure-client/go-span-profiles.md [hunk 5]
```diff
@@ -7,38 +7,52 @@ weight: 100
...
 
+Span Profiles represents a major shift in profiling methodology, enabling deeper analysis of both tracing and profiling data.
+Traditional continuous profiling provides a system-wide view over fixed intervals.
```
Comment (kolesnikovae) [inline docs/sources/configure-client/go-span-profiles.md:11]: Nit: system-wide or application-wide

**Review by kolesnikovae — APPROVED** [review]
LGTM

File: docs/sources/configure-client/go-span-profiles.md (hunk 5 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/go-span-profiles.md:11]: Which one should it be? Application-wide?

**Comment by knylander-grafana** [comment]
@kolesnikovae Thank you for the review! Should the wording for your comment above be application-wide or system-wide?

File: docs/sources/configure-client/go-span-profiles.md (hunk 5 above)
Comment (kolesnikovae) [reply docs/sources/configure-client/go-span-profiles.md:11]: Now that I've thought a bit more, I'm not sure if we need to clarify: I can interpret "system-wide" in a broader sense that also covers the "application-wide" case. Sorry, I didn't mean to confuse you :)

In Pyroscope, we have a strong concept of a "service" and encourage users to adhere to it. So if I had to choose, I'd pick "application-wide."

Comment (knylander-grafana) [reply docs/sources/configure-client/go-span-profiles.md:11]: Okay! If it is clearer to say application-wide I can change it. Although, your comment makes me wonder if I should say "system-wide and appilcation-wide"...

Comment (knylander-grafana) [reply docs/sources/configure-client/go-span-profiles.md:11]:
```suggestion
Traditional continuous profiling provides an application-wide view over fixed intervals.
```

**Pull request merged**